lumicks/pylake/kymotracker/kymoline.py

```diff
@@ -83,50 +83,72 @@ def import_kymolinegroup_from_csv(filename, image, delimiter=";"):
 class KymoLine:
     """A line on a kymograph"""
 
-    __slots__ = ["time_idx", "coordinate_idx", "image_data"]
+    __slots__ = ["time_idx", "coordinate_idx", "_image"]
 
-    def __init__(self, time_idx, coordinate_idx, image_data=None):
+    def __init__(self, time_idx, coordinate_idx, image):
+        """A kymograph line.
+
+        time_idx : array_like
+            List of time indices [pixels].
+        coordinate_idx : array_like
+            List of coordinate indices [pixels].
+        image : CalibratedKymographChannel
+            Calibrated image data.
+        """
         self.time_idx = list(time_idx)
         self.coordinate_idx = list(coordinate_idx)
-        self.image_data = image_data
+        self._image = image
 
     def append(self, time_idx, coordinate_idx):
-        """Append time, coordinate pair to the KymoLine"""
+        """Append time [pixels], coordinate pair [pixels] to the KymoLine"""
         self.time_idx.append(time_idx)
         self.coordinate_idx.append(coordinate_idx)
 
     def with_offset(self, time_offset, coordinate_offset):
         """Returns an offset version of the KymoLine"""
+        # Convert from image units to pixels
+        time_offset = self._image.from_seconds(time_offset)
+        coordinate_offset = self._image.from_coord(coordinate_offset)
+
         return KymoLine(
             [time_idx + time_offset for time_idx in self.time_idx],
             [coordinate_idx + coordinate_offset for coordinate_idx in self.coordinate_idx],
-            self.image_data,
+            self._image,
         )
 
     def __add__(self, other):
         """Concatenate two KymoLines"""
         return KymoLine(
             self.time_idx + other.time_idx,
             self.coordinate_idx + other.coordinate_idx,
-            self.image_data,
+            self._image,
         )
 
     def __getitem__(self, item):
         return np.squeeze(
             np.array(np.vstack((self.time_idx[item], self.coordinate_idx[item]))).transpose()
         )
 
+    @property
+    def time(self):
+        return [self._image.to_seconds(t) for t in self.time_idx]
+
+    @property
+    def coordinate(self):
+        return [self._image.to_coord(t) for t in self.coordinate_idx]
```

Comment:
It might be nice to return these as `np.array` rather than a regular `list`.

Also, maybe my own weird quirk, but I would change the iterated variable to `x` instead of `t` in `coordinate`

Reply:
Sure, will do!